Runner/Sandbox.cs

```diff
@@ -184,26 +167,19 @@ private void SetAppConfigIfExists()
 
         private void EnumerateAndLoadAssemblies()
         {
-            ScannedAssemblies = Directory.EnumerateFiles(Utils.GetGaugeBinDir(), "*.dll", SearchOption.TopDirectoryOnly)
-                .Select(s =>
-                {
-                    Logger.Debug("Loading assembly from : {0}", s);
-                    return Assembly.Load(AssemblyName.GetAssemblyName(s));
-                })
-                .ToList();
+            foreach (var path in Directory.EnumerateFiles(Utils.GetGaugeBinDir(), "*.dll", SearchOption.TopDirectoryOnly))
+                AssemblyScanner.TryAdd(path);
         }
 
         private void LocateTargetLibAssembly()
         {
-            TargetLibAssembly = ScannedAssemblies.First(assembly => assembly.GetName().Name == GaugeLibAssembleName);
+            TargetLibAssembly = typeof(Step).Assembly;
```

Comment:
I see what you mean, fair enough. 

The `Unique` call would end up comparing to `Assembly` types, perhaps we could use 

```
AssemblyName targetLibName = everythingFromGaugeBin.GetReferencedAssemblies()
.First(a => a.Name == typeof(Step).Assembly.Name);
```